vm/src/org/mozartoz/truffle/translator/Translator.java

```diff
@@ -531,36 +535,44 @@ private Arity buildArity(OzArity arity) {
 	}
 
 	private OzNode translateBinaryOp(String operator, OzNode left, OzNode right) {
-		left = deref(left);
-		right = deref(right);
 		switch (operator) {
+		case "==":
+			return createBinaryOp(EqualNodeFactory.getInstance(), left, right);
+		case "\\=":
+			return createBinaryOp(NotEqualNodeFactory.getInstance(), left, right);
 		case "+":
-			return AddNodeFactory.create(left, right);
+			return createBinaryOp(AddNodeFactory.getInstance(), left, right);
 		case "-":
-			return SubNodeFactory.create(left, right);
+			return createBinaryOp(SubNodeFactory.getInstance(), left, right);
 		case "*":
-			return MulNodeFactory.create(left, right);
+			return createBinaryOp(MulNodeFactory.getInstance(), left, right);
+		case "/":
+			return createBinaryOp(FloatDivNodeFactory.getInstance(), left, right);
 		case "div":
-			return DivNodeFactory.create(left, right);
+			return createBinaryOp(DivNodeFactory.getInstance(), left, right);
 		case "mod":
-			return ModNodeFactory.create(left, right);
-		case "==":
-			return EqualNodeFactory.create(left, right);
-		case "\\=":
-			return NotEqualNodeFactory.create(left, right);
+			return createBinaryOp(ModNodeFactory.getInstance(), left, right);
 		case "<":
-			return LesserThanNodeFactory.create(left, right);
+			return createBinaryOp(LesserThanNodeFactory.getInstance(), left, right);
 		case "=<":
-			return LesserThanOrEqualNodeFactory.create(left, right);
+			return createBinaryOp(LesserThanOrEqualNodeFactory.getInstance(), left, right);
 		case ">":
-			return GreaterThanNodeFactory.create(left, right);
+			return createBinaryOp(GreaterThanNodeFactory.getInstance(), left, right);
+		case ">=":
+			return createBinaryOp(GreaterThanOrEqualNodeFactory.getInstance(), left, right);
 		case ".":
-			return DotNodeFactory.create(left, right);
+			return createBinaryOp(DotNodeFactory.getInstance(), left, right);
+		case ":=":
+			return createBinaryOp(CatExchangeNodeFactory.getInstance(), left, right);
 		default:
 			throw unknown("operator", operator);
 		}
 	}
 
+	private static OzNode createBinaryOp(NodeFactory<? extends OzNode> factory, OzNode left, OzNode right) {
+		return BuiltinsManager.createNodeFromFactory(factory, new OzNode[] { left, right });
```

Comment:
Just as a note: you could keep the varargs signature, it also accepts an OzNode[] as a single final argument in Java.